src/dex-ui/store/poolsSlice/poolsSlice.ts

```diff
@@ -122,6 +201,44 @@ const createPoolsSlice: PoolsSlice = (set, get): PoolsStore => {
       }
       app.setFeaturesAsLoaded(["allPoolsMetrics"]);
     },
+    fetchSpotPrices: async ({ inputTokenAddress, outputTokenAddress, contractId }: FetchSpotPriceParams) => {
```

Comment:
We should modify the `fetchSpotPrices` function on the `swap` slice instead of creating a duplicate function.

Reply:
we can discuss on this as for now we we fetch spot prices for selected pair (user has to select a pair for it get fetched), if the user goes to `AddLiquidity` Page without doing anything on `swap` page the in add liquidity there will be no way to get it, so thats why this function is in both `swapSlice` and `poolSlice`

Reply:
Ok, we can update this in a future iteration. We're currently managing selected token accounts and routes in the Zustand store for swapping. We should aim to move this data and logic to React UI / React Query state.